### PR TITLE
Fixing 9 -> 10 Upgrade Path (From legacy chart). Numerous small fixes to Chart.

### PR DESCRIPTION
**Pull request by DandyDeveloper**
Fixes include: 
 - Better Broker and Redis support with dependencies
 - Using official Kafka repository and settings
 - Increase worker count to 3 for typical Sentry workload
 - More inline changes with postgresql dependency expectations.

**Problems**
Right now, biggest issue is without a Helm plugin, there's no nice way to control the installation of all these components, so you end up with a couple of pod restarts (max 2 in my experience) before everything comes up successfully. 

I've confirmed and on an initial install. This works without problems, I am testing the upgrade path and may have some additional minor updates required.

### CODE REVIEW

**Comment by DandyDeveloper** [comment]
@Mokto I've made a bunch of updates here. It's a lot to take in. I've noted the key things in the description.

**Comment by Mokto** [comment]
Can you please leave the chart dependencies as zip files, as it should ?

It is also going going easier for me to read you PR.

Thanks

**Comment by DandyDeveloper** [comment]
@Mokto I have added them as requested - Quick question: Why do you want them in there? Seems like unnecessary data in the repo?

**Comment by Mokto** [comment]
Ah, thanks! You're right, you can remove them if you want. 
I was just struggling with so many charts files ;)

I was also thinking, maybe it's time to include :
- redis as highly available (through redis-ha + haproxy)
- kafka as highly available. (See https://github.com/sentry-kubernetes/charts/pull/26)

I'm aware that the current stable chart is using `redis`, not `redis-ha`, but for a migration, it's still ok, I guess.

I would put the whole sentry stack as highly available.

What do you think ?

**Comment by DandyDeveloper** [comment]
@mokto I own the redis-ha chart so, why not :). https://github.com/dandydeveloper/charts

As for Kafka, maybe in a separate PR? Not that familiar with the Kafka architecture for ha. But I definitely think it's a good idea.

**Comment by Mokto** [comment]
Ok, let's do that then. 

What you've done is really nice, but it's a big change for anyone using the repo.
So let's do all the big changes at once.

And I'll change Kafka after your pull request is merged.

**Comment by Mokto** [comment]
By the way, have you tested the upgrade path?

**Comment by DandyDeveloper** [comment]
@Mokto Not fully, testing edge cases that I can think of.

**Comment by DandyDeveloper** [comment]
@Mokto Having some issues with the Redis upgrades, looks like it is trying to update parts of the statefulset. Needs a little more time.

**Comment by DandyDeveloper** [comment]
@Mokto All done, I've tested a few use cases, added a README for the edge cases basically. Overall, it works a treat. 

I'd worry about HA redis and PSQL outside of this, because right now, with the helm 2 to 3 being so jumpy, it makes upgrading several components really frustrating.

For some reason, linting is failing... Not sure why?

**Comment by Mokto** [comment]
Still in the process of checking.

I've made a few changes to support external databases.
And fix the lint issue.

**Comment by Mokto** [comment]
I'm almost done ;) I'll merge your PR soon (thanks!). And I'll create a PR immediately after to fix the small issues.

File: sentry/values.yaml [hunk 1]
```diff
@@ -107,54 +98,33 @@ hooks:
...
 
 system:
-  url: ''
-  adminEmail: ''
+  url: ""
+  adminEmail: ""
   secretKey: 'icLq77rCyY_qrMMpXa6TQNjkDV6mU!c'
   public: false #  This should only be used if you’re installing Sentry behind your company’s firewall.
 
 mail:
```
Comment (gabemeola) [inline sentry/values.yaml:106]: Should this key stay `email` for interop with the old chart?

File: sentry/values.yaml [hunk 2]
```diff
@@ -245,6 +215,81 @@ filestore:
 
 config:
   yaml: |
-    # YAML
```
Comment (gabemeola) [inline sentry/values.yaml:248]: Similar question. Old chart calls this `configYml` and `sentryConfPy`

File: sentry/values.yaml [hunk 3]
```diff
@@ -107,54 +98,33 @@ hooks:
...
-  db: 0
-
 kafka:
   host: kafka-confluent
   port: 9092
 
 service:
   name: sentry
-  type: LoadBalancer
+  type: ClusterIP
```
Comment (gabemeola) [inline sentry/values.yaml:127]: Curious, why change from `LoadBalancer`?

Comment (i5okie) [reply sentry/values.yaml:127]: Not everybody uses LoadBalancer. It's an expensive option. I think it's more common to see some sort of an ingress controller being used with service set to ClusterIP. IMO anyway.

Comment (gabemeola) [reply sentry/values.yaml:127]: Makes sense thanks for clarifying

File: sentry/values.yaml (hunk 1 above)
Comment (Mokto) [reply sentry/values.yaml:106]: Thanks for your report. I'll take care of that in the next PR

File: sentry/values.yaml (hunk 2 above)
Comment (Mokto) [reply sentry/values.yaml:248]: Thanks for your report. I'll take care of that in the next PR

File: sentry/values.yaml (hunk 1 above)
Comment (Mokto) [reply sentry/values.yaml:106]: Actually this one is not the previous chart.

File: sentry/values.yaml (hunk 2 above)
Comment (DandyDeveloper) [reply sentry/values.yaml:248]: The good thing is upgrade paths deal with this. It's less of an issue as config maps are different anyway. So it's not a huge problem for people to change the values.

**Pull request merged**